### PR TITLE
Add GATConv support for HeteroGraphConv 

### PR DESCRIPTION
**Pull request by rbSparky**
Covers Issue #311 

Needed to change `softmax_edge_neighbors` to accommodate hetero graphs

### CODE REVIEW

File: src/layers/conv.jl [hunk 1]
```diff
@@ -409,25 +409,31 @@ end
...
     @assert !((e === nothing) && (l.dense_e !== nothing)) "Input edge features required for this layer"
     @assert !((e !== nothing) && (l.dense_e === nothing)) "Input edge features were not specified in the layer constructor"
 
+    xj, xi = expand_srcdst(g, x)
+    edge_t = g isa GNNHeteroGraph ? g.etypes[1] : nothing
+
     if l.add_self_loops
         @assert e===nothing "Using edge features and setting add_self_loops=true at the same time is not yet supported."
-        g = add_self_loops(g)
+        g = g isa GNNHeteroGraph ? add_self_loops(g, edge_t) : add_self_loops(g)
```
[the remark this answers is not in the document]
Comment (rbSparky) [reply src/layers/conv.jl]: I'll edit `add_self_loops` in the next PR and change `conv.jl` to use it wherever this branching is currently being used. 
For now I'll let the `if` statement be for this PR.

File: src/layers/conv.jl [hunk 2]
```diff
@@ -409,25 +409,31 @@ end
...
     _, chout = l.channel
     heads = l.heads
 
-    Wx = l.dense_x(x)
-    Wx = reshape(Wx, chout, heads, :)                   # chout × nheads × nnodes
+    Wxj = l.dense_x(xj)
+    Wxj = reshape(Wxj, chout, heads, :)                   
+
+    Wxi = l.dense_x(xi)
+    Wxi = reshape(Wxi, chout, heads, :)
```
[the remark this answers is not in the document]
Comment (rbSparky) [reply src/layers/conv.jl]: Done

**Comment by rbSparky** [comment]
tests failing will check

File: src/layers/conv.jl [hunk 3]
```diff
@@ -409,25 +409,37 @@ end
...
 
-function (l::GATConv)(g::GNNGraph, x::AbstractMatrix,
+function (l::GATConv)(g::AbstractGNNGraph, x,
                       e::Union{Nothing, AbstractMatrix} = nothing)
     check_num_nodes(g, x)
     @assert !((e === nothing) && (l.dense_e !== nothing)) "Input edge features required for this layer"
     @assert !((e !== nothing) && (l.dense_e === nothing)) "Input edge features were not specified in the layer constructor"
 
+    xj, xi = expand_srcdst(g, x)
+    edge_t = g isa GNNHeteroGraph ? g.etypes[1] : nothing
```
Comment (CarloLucibello) [inline src/layers/conv.jl:419]:
```suggestion
```

Comment (rbSparky) [reply src/layers/conv.jl]: Done

**Pull request merged**